src/hotspot/share/runtime/arguments.cpp

```diff
@@ -3449,6 +3449,11 @@ void Arguments::set_shared_spaces_flags_and_archive_paths() {
   if (DumpSharedSpaces || UseSharedSpaces) {
     init_shared_archive_paths();
   }
+
+  if (DumpSharedSpaces && !UseCompressedOops) {
+    FLAG_SET_CMDLINE(MaxHeapSize, (size_t)4 * G);
```

Comment:
I think we should override MaxHeapSize only if it's greater than 4G. The user should be able to choose a smaller value.

Reply:
Per our off list discussions, the InitialHeapSize, MinHeapSize, and MaxHeapSize are being set in
`universe_init()` via `MetaspaceShared::adjust_heap_sizes_for_dumping()`.